### PR TITLE
Week 6 Slack API project requirements

### PR DESCRIPTION
**Pull request by droberts-sea**
## Description
I followed the [project template](https://github.com/Ada-Developers-Academy/Teachers-Edition-Textbook/blob/master/project-template.md) from the TET. This repo is pretty straight-ahead - not much scaffolding, the only complex bit in the README is setting up Slack auth.

Dan's solution lives at https://github.com/droberts-ada/c11-week6-project

## Todos
- [ ] In-class design activity
- [ ] Figure out some sort of way to demo the project without giving them source code

## Feedback Requested
- [ ] Run through the Slack setup process - is my writeup good enough? Should this be here or in a separate document?
- Find my dumb typos

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,161 @@
-# slack-cli
-Practice working with APIs by writing a command line application that interacts with the Slack API
+# Project Title
```
Comment (kaidamasaki) [inline README.md]: Are they supposed to put their project title here?  Or is this a placeholder for instructors?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,161 @@
...
+
+**Hint:** You may want to investigate the [Table Print gem](https://github.com/arches/table_print) to handle formatting tables.
+
+### Wave 2 - Selecting and Showing Details
+
+**Vocab:** channels and users are both _recipients_.
+
+As a user at the input prompt...
+
+- I should see three new options:
```
Comment (tildeee) [inline README.md]: nitpicking: could we add in the word "additional" somewhere here? (To clarify that these threw new options do not _replace_ the previous ones)

File: README.md [hunk 3]
```diff
@@ -1,2 +1,161 @@
...
+- When I type "select channel", I should be able to supply a channel name or Slack ID. The corresponding channel should be the "selected" recipient.
+- For selecting both users and channels, if no user/channel has that name or ID, the program should let me know and return to the main command loop.
+- When I type "details", the program should print out details for the currently selected recipient. What information is printed depends on whether it's a channel or a user.
+  - If no recipient is currently selected, the program should let me know and return to the main command prompt.
+
+### Wave 3 - Sending Messages
+
+As a user at the input prompt...
+
+- I should see one new option: send message.
```
Comment (tildeee) [inline README.md]: Same as above re: "additional" new option

File: README.md [hunk 4]
```diff
@@ -1,2 +1,161 @@
...
+- When I type "send message", if a recipient is selected the program should ask me to type out a message, which will be sent to that recipient.
+  - If no recipient is currently selected, the program should let me know and return to the main command prompt.
+
+## Optional Enhancements
+
+Don't even think about reading these until all the primary requirements are complete.
+
+- As a user, I can change global settings for the program
+  - These settings include the username for the bot and an emoji to use as an icon
+  - When I change these settings, the program should save them in the JSON format in a file named `bot-settings.json`. When I restart the program, it should reload those settings.
```
Comment (tildeee) [inline README.md:172]: by this point, have we given them a good resource to think about reading a json file? I'm assuming this optional is really "above and beyond," so I want to confirm.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,161 @@
...
+### Wave 2 - Selecting and Showing Details
+
+**Vocab:** channels and users are both _recipients_.
+
+As a user at the input prompt...
+
+- I should see three new options:
+  - Select user
+  - Select channel
+  - Details
```
Comment (tildeee) [inline README.md]: just to help clarify that there is one recipient that is either a user or a channel, could we make the options something like:
```
- Select a user as the recipient
- Select a channel as the recipient
- List recipient
```
or something like that?

**Review by tildeee — CHANGES_REQUESTED** [review]
Overall this looks great! My random thoughts that don't fit as code comments are:

- My biggest thing which is why I have this as "Request changes": The requirements around testing for this project seem unclear to me-- in what capacity they're required
- The comprehension questions are huge! Which is not a bad thing, and I can't think of which of those questions I would cut, but I just wanted to bring that up.
- Still waiting on your TODOs :)

Otherwise, looks like a great project! I'm into it. I think it's a great 5-day pair project.

File: README.md (hunk 4 above)
Comment (droberts-sea) [reply README.md:172]: We have not. The intent here is to give something really meaty to work on.

**Review by tildeee — APPROVED** [review]

**Review by tildeee — APPROVED** [review]
Still looking for that in-design class activity, but assuming that goes well, this is an "approve" from me

File: README.md [hunk 6]
```diff
@@ -1,2 +1,178 @@
...
+- Pair, [stage 2](https://github.com/Ada-Developers-Academy/pedagogy/blob/master/rule-of-three.md#stage-2) project
+- Due EOD Friday, *DATE HERE*
+
+## Introduction
+
+To practice working with APIs, we will utilize the API for a tool with which you already very familiar: Slack!
+
+The application we build this week will be designed for humans to use. This means the format will be familiar to us, but it seems a little silly - why would you use the command line when you could just use the app?
+
+However, there are many places where you might want to use slack, where it doesn't make sense to have a human in the loop. For example, you might set up a continuous integration tool like [Travis CI](https://travis-ci.org/) to run your tests automatically whenever you submit a pull request. Travis has the ability to post in your Slack channel when the tests are done, using the Slack API in a way similar to the way we will in this project.
```
Comment (kaidamasaki) [inline README.md]:
```suggestion
However, there are many places where you might want to use Slack, where it doesn't make sense to have a human in the loop. For example, you might set up a continuous integration tool like [Travis CI](https://travis-ci.org/) to run your tests automatically whenever you submit a pull request. Travis has the ability to post in your Slack channel when the tests are done, using the Slack API in a way similar to the way we will in this project.
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,178 @@
...
+    ![Authorize the App](images/slack-auth-setup-authorize.png)
+1. You should now see an `OAuth Access Token`. Copy this and paste it into your app's `.env` file.
+    ![OAuth Token](images/slack-auth-setup-token.png)
+    - The `.env` file should already be in the provided `.gitignore`, but it doesn't hurt to double check!
+
+Your application should now be able to access Slack!
+
+### Verification
+
+To verify that you've set up the slack token correctly, work with your partner to write a script that does the following:
```
Comment (kaidamasaki) [inline README.md]:
```suggestion
To verify that you've set up the Slack token correctly, work with your partner to write a script that does the following:
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,178 @@
...
+
+1. Use the `dotenv` gem to load environment variables
+1. Use HTTParty to send a GET request to the [`channels.list` endpoint](https://api.slack.com/methods/channels.list)
```
Comment (kaidamasaki) [inline README.md]: Consider

```suggestion
1. Use HTTParty to send a `GET` request to the [`channels.list` endpoint](https://api.slack.com/methods/channels.list)
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,178 @@
...
+
+As a user at the input prompt...
+
+- I should see one additional option: `send message`.
+- When I type `send message`, if a recipient is selected the program should ask me to type out a message, which will be sent to that recipient.
+  - If no recipient is currently selected, the program should let me know and return to the main command prompt.
+
+## Optional Enhancements
+
+Don't even think about reading these until all the primary requirements are complete.
```
Comment (kaidamasaki) [inline README.md]:
```suggestion
Don't even think about reading these until all the primary requirements are complete!
```

**Review by kaidamasaki — APPROVED** [review]
This project is really cool, and looks great!

**Pull request merged**